other/save-generator.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "../toxcore/tox.h"
+#include "../testing/misc_tools.h"
+#include "../toxcore/ccompat.h"
+
+#define SAVE_FILE "save.tox"
+#define STATUS_MESSAGE "Hello World"
+
+static const char *bootstrap_ip = "185.14.30.213";
+static const char *bootstrap_address = "2555763C8C460495B14157D234DD56B86300A2395554BCAE4621AC345B8C1B1B";
+static const uint16_t udp_port = 443;
+
+static bool write_save(const uint8_t *data, size_t length)
+{
+    FILE *fp = fopen(SAVE_FILE, "w");
+
+    if (!fp) {
+        return false;
+    }
+
+    if (fwrite(data, length, 1, fp) != 1) {
+        fclose(fp);
+        return false;
+    }
+
+    fclose(fp);
+    return true;
+}
+
+int main(int argc, char *argv[])
+{
+    if (argc < 3) {
+        printf("Usage: ./save-generator <name> <friend id>\n");
+        return -1;
+    }
+
+    Tox *tox = tox_new(nullptr, nullptr);
+
+    if (!tox) {
+        printf("Failed to create tox.\n");
+        return -1;
+    }
+
+    uint8_t *key = hex_string_to_bin(bootstrap_address);
+
+    if (!key) {
+        printf("Could not allocate memory for tox address\n");
+        return -1;
+    }
+
+    tox_bootstrap(tox, bootstrap_ip, udp_port, key, nullptr);
+    free(key);
+
+    while (true) {
+        tox_iterate(tox, nullptr);
+
+        if (tox_self_get_connection_status(tox) == TOX_CONNECTION_UDP) {
```

Comment:
Prefer callbacks. This main function is quite long. Given that we'll want to extend it in the future, it's probably good to split it into a few functions.